src/signTx.c

```diff
@@ -235,6 +238,20 @@ static inline void advanceStage()
 			// we are not waiting for any APDU here, network id is already known from the init APDU
 			txHashBuilder_addNetworkId(&BODY_CTX->txHashBuilder, ctx->commonTxData.networkId);
 		}
+		ctx->stage = SIGN_STAGE_BODY_TOTAL_COLLATERAL;
+
+	case SIGN_STAGE_BODY_TOTAL_COLLATERAL:
+		if (ctx->includeTotalCollateral) {
+			txHashBuilder_addTotalCollateral(&BODY_CTX->txHashBuilder, ctx->txColl);
```

Comment:
I don't think we have received txColl yet. I'd expect we would do something analogous to what we do with fees (sent in a separate APDU).